src/matrix/kernelmatrix.jl

```diff
@@ -100,6 +100,16 @@ function kernelmatrix(κ::SimpleKernel, x::AbstractVector, y::AbstractVector)
     return map(d -> kappa(κ, d), pairwise(metric(κ), x, y))
 end
 
+function kernelmatrix_diag(κ::SimpleKernel, x::AbstractVector)
+    return map(d -> kappa(κ, d), colwise(metric(κ), x))
```

Comment:
BTW one could use `Base.Fix1` here instead of the custom closure:
```suggestion
    return map(Base.Fix1(kappa, κ), colwise(metric(κ), x))
```

Reply:
When was `Fix1` introduced? I thought it was a recent addition

Reply:
It exists since Julia 0.7.0-alpha: https://github.com/JuliaLang/julia/pull/26708

Reply:
Haha while watching the Julia PR I saw this PR reference appearing :p